### PR TITLE
DOP-3948

### PR DESCRIPTION
**Pull request by seungpark**
[DOP-3948](https://jira.mongodb.org/browse/DOP-3948)

**Goal:** Convert page documents from parser into manifest entries for search.documents[], namely the "facets" field

Example of output page documents from parser is included in the ticket.
Updated bson includes an update to the expected page document:
```
{
	"page_id": "cloud-docs/seung.park/DOP-3851-b-test-facet-directives/reference/google-gcp",
	"filename": "reference/google-gcp.txt",
	"ast": {...},
	"source": "",
	"static_assets": [...],
	"facets": [
		{
			"category": "programming_language",
			"value": "cpp",
			"sub_facets": null
		},
		{
			"category": "genre",
			"value": "reference",
			"sub_facets": null
		},
		{
			"category": "target_product",
			"value": "atlas",
			"sub_facets": [
				{
					"category": "sub_product",
					"value": "search",
					"sub_facets": null
				}
			]
		}
	]
}

```

### CODE REVIEW

**Review by i80and — COMMENTED** [review]
Just a couple syntax notes

File: mut/index/SnootyManifest.py [hunk 1]
```diff
@@ -125,13 +145,41 @@ def derive_preview(self) -> Optional[str]:
...
+            document_facets[key] = document_facets.get(key, [])
+            document_facets[key].append(facet.value)
+            if not facet.sub_facets:
+                return
+            for sub_facet in facet.sub_facets:
+                insert_key_values(
+                    sub_facet, key + HIERARCHY_KEY + facet.value + HIERARCHY_KEY
+                )
+
+        def create_facet(facet_entry: any):
```
[the remark this answers is not in the document]
Comment (seungpark) [reply mut/index/SnootyManifest.py:166]: woops! updated this and the constructor to objects

File: mut/index/SnootyManifest.py [hunk 2]
```diff
@@ -12,6 +12,27 @@
 logger = logging.getLogger(__name__)
 
 
+class Facet:
+    category: str
+    value: str
+    sub_facets: Optional[List["Facet"]] = None
+
+    def __init__(self, category: str, value: str, sub_facets: [List[any]]) -> None:
```
[the remark this answers is not in the document]
Comment (seungpark) [reply mut/index/SnootyManifest.py]: removed! changed to object

File: mut/index/SnootyManifest.py [hunk 3]
```diff
@@ -12,6 +12,27 @@
 logger = logging.getLogger(__name__)
 
 
+class Facet:
+    category: str
```
[the remark this answers is not in the document]
Comment (seungpark) [reply mut/index/SnootyManifest.py]: removed. thanks for the note! had written this originally with @dataclass decorator but was not working as intended

**Review by rayangler — APPROVED** [review]
LGTM!

**Pull request merged**